### PR TITLE
Use CircleCI to paralellize the build.

### PR DESCRIPTION
**Pull request by SuperQ**
Use a CircleCI matrix param to parallelize the build.
* Update build-bumper to include circleci config
* Use docker image save trick to persist docker images between build and
  publish.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
Still needs some work, but getting closer.

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,21 +10,33 @@ jobs:
...
+    parameters:
+      version:
+        type: string
+
     steps:
     - checkout
     - setup_remote_docker
     - run: docker info
-    - run: make
+    - run: make VARIANTS=base VERSION=<< parameters.version >>
```
Comment (roidelapluie) [inline .circleci/config.yml:21]:
```suggestion
    - run: make VERSION=<< parameters.version >>
```

File: Makefile [hunk 2]
```diff
@@ -14,21 +14,20 @@
...
+	@./build.sh "$(VERSION)" "$(VARIANTS)"
 
 tag:
-	docker tag "$(REPOSITORY)/$(NAME):$(LATEST)-main" "$(REPOSITORY)/$(NAME):latest"
-	docker tag "$(REPOSITORY)/$(NAME):$(LATEST)-main" "$(REPOSITORY)/$(NAME):main"
-	docker tag "$(REPOSITORY)/$(NAME):$(LATEST)-main" "$(REPOSITORY)/$(NAME):arm"
-	docker tag "$(REPOSITORY)/$(NAME):$(LATEST)-main" "$(REPOSITORY)/$(NAME):powerpc"
-	docker tag "$(REPOSITORY)/$(NAME):$(LATEST)-main" "$(REPOSITORY)/$(NAME):mips"
-	docker tag "$(REPOSITORY)/$(NAME):$(LATEST)-main" "$(REPOSITORY)/$(NAME):s390x"
+	docker tag "$(REPOSITORY)/$(NAME):$(VERSION)-main" "$(REPOSITORY)/$(NAME):latest"
```
Comment (roidelapluie) [inline Makefile:25]: We can only do this if VERSION is latest

File: .circleci/config.yml (hunk 1 above)
Comment (SuperQ) [reply .circleci/config.yml:21]: I'm doing this intentionaly to split base and main into two steps.

File: Makefile (hunk 2 above)
Comment (SuperQ) [reply Makefile:25]: VERSION will always be the latest when tag is called. We only call `make tag` from the publish workflow, not the build workflow. `make all` does not call `tag`.

**Comment by SuperQ** [comment]
I'm also thinking about completely eliminating the versioned directories, and moving to a template-based Dockerfile setup.

Something like [this Dockerfile.j2](https://gitlab.com/toertel/docker-image-logitech-media-server/-/blob/master/Dockerfile.j2).

**Comment by roidelapluie** [comment]
Is this still WIP?

**Comment by SuperQ** [comment]
Ok, fixed the image saving in the workflow, this should publish correctly now.

**Comment by SuperQ** [comment]
Ping

**Review by roidelapluie — APPROVED** [review]
Pong :)

**Pull request merged**